### ISSUE TITLE
Won't override existing Go version properly

### ISSUE THREAD

**Issue by mvdan** [travis-ci/gimme#90]
On CircleCI (I know, I know) they ship a Go version with the system, which is 1.8 at the time of writing. We use `gimme` to override that with 1.7.5 or whatever other version we actually need.

This worked well until this week, because Circle used to ship 1.7.x and now ships 1.8.x. Then or builds get confused by Go version mismatches:

```
$ eval "$(gimme 1.7)" && cd $PROJECT_PATH && ./bin/ci-tests.sh
go version go1.7 linux/amd64
# testmain
foo:82: cannot use matchString (type func(string, string) (bool, error)) as type testing.testDeps in argument to testing.MainStart:
	func(string, string) (bool, error) does not implement testing.testDeps (missing MatchString method)
```

Presumably, this is because the `go` binary is from 1.7, while `GOROOT` points at 1.8. I confirmed this by changing the shell line to `unset GOROOT && eval "$(gimme 1.7)" && cd $PROJECT_PATH && ./bin/ci-tests.sh`, which fixed it.

The fix is not perfectly clear to me, but it sounds like `gimme` is not unsetting or setting `GOROOT` properly in its output to be evaluated.

**Comment by mvdan** [comment]
Actually, the `unset GOROOT` didn't fix it.

**Comment by mvdan** [comment]
Still haven't found the cause of this, but doesn't seem like a `gimme` issue for now. Apologies for the noise.

**Comment by mvdan** [comment]
Fixed the issue finally - it was #87.
